tests/integration/components/o-s-s/avatar-test.ts

```diff
@@ -0,0 +1,98 @@
+      await render(hbs`<OSS::Avatar @image="http://foo.co/bar.png" />`);
+
+      assert.dom('.upf-avatar').exists();
+      assert.dom('.upf-avatar img').exists();
+      assert.dom('.upf-avatar img').hasAttribute('src', 'http://foo.co/bar.png');
+    });
+
+    test('it displays the initials when initials are provided', async function (assert) {
+      await render(hbs`<OSS::Avatar @initials="TS" />`);
+
+      assert.dom('.upf-avatar').exists();
+      assert.dom('.upf-avatar span').exists();
+      assert.dom('.upf-avatar span').hasText('TS');
+    });
+
+    test('it displays the default image when initials and image are not provided', async function (assert) {
+      await render(hbs`<OSS::Avatar />`);
+
+      assert.dom('.upf-avatar').exists();
+      assert.dom('.upf-avatar img').exists();
+      assert.dom('.upf-avatar img').hasAttribute('src', DEFAULT_IMAGE_URL);
+    });
+
+    test('it displays the image when image and initials are provided', async function (assert) {
+      await render(hbs`<OSS::Avatar @image="http://foo.co/bar.png" @initials="TS" />`);
+
+      assert.dom('.upf-avatar').exists();
+      assert.dom('.upf-avatar img').exists();
+      assert.dom('.upf-avatar img').hasAttribute('src', 'http://foo.co/bar.png');
+    });
+  });
+
+  module('Sizes', function () {
+    test('it sets the right default class when size is not provided', async function (assert) {
+      await render(hbs`<OSS::Avatar />`);
+
+      assert.dom('.upf-avatar').exists();
+      assert.dom('.upf-avatar').hasClass(`upf-avatar--md`);
+    });
+
+    Object.keys(SizeDefinition).forEach((size) => {
+      test(`it sets the right class when using a supported size: ${size}`, async function (assert: Assert) {
+        this.size = size;
+        await render(hbs`<OSS::Avatar @size={{this.size}} />`);
+
+        assert.dom('.upf-avatar').exists();
+        assert.dom('.upf-avatar').hasClass(`upf-avatar--${size}`);
+      });
+    });
+  });
+
+  module('Error behavior', function () {
+    test('it throws an error if wrong size argument is passed', async function (assert: Assert) {
+      setupOnerror((err: Error) => {
+        assert.equal(
+          err.message,
+          `Assertion Failed: [component][OSS::Avatar] Unknown size. Available sizes are: ${Object.keys(
+            SizeDefinition
+          ).join(', ')}`
+        );
+      });
+
+      await render(hbs`<OSS::Avatar @size="test" />`);
+    });
+
+    test('it displays the initials when initials and image are provided but image loading get an error', async function (assert) {
+      await render(hbs`<OSS::Avatar @image="http://foo.co/bar.p" @initials="TS" />`);
+      await waitFor('.upf-avatar span');
+
+      assert.dom('.upf-avatar').exists();
+      assert.dom('.upf-avatar span').exists();
+      assert.dom('.upf-avatar span').hasText('TS');
+    });
+
+    test('it displays the default image when image is provided but image loading get an error', async function (assert) {
```

Comment:
`It displays the placeholder image when the image provided in parameters fails to load`